test/230_dns_unqualified_test.sh

```diff
@@ -26,4 +26,8 @@ check c2
 check c3
 check "$container"
 
+# check that unqualified names are automatically qualified when looking up
+weave_on $HOST1 dns-add $C2 c2 -h name1.$DOMAIN
+assert_dns_a_record $HOST1 c2 name1 $C2 name1.$DOMAIN
+
```

Comment:
Is this test actually failing on master? I wouldn't have thought so, since the image we use in `start_container_with_dns` has a functioning resolver library and hence will append the domain to the query.

I also don't understand what the point is of adding another record and extending the `assert_dns_record` helper function, instead of using `check` defined above.

So, imo, this should just start another container, `c5`, with a broken resolver library (`start_container` will do that), and run `check c5`. That *should* fail on master and succeed here.